docs/website/mtouch-errors.md

```diff
@@ -918,6 +978,14 @@ To fix this warning, open the project file in a text editor, and remove all `Mto
 
 This means there is an error on the [custom XML linker configuration file](https://developer.xamarin.com/guides/cross-platform/advanced/custom_linking/) you provided, please review your file.
 
+<h3><a name="MT2018"/>MT2018: The assembly '*' is referenced from two different locations: '*' and '*'.</h3>
+
+The assembly mentioned in the error message is loaded from multiple locations. Make sure to always use the same version of an assembly.
+
```

Comment:
This error will not be shown when the same assembly is referenced from different paths between app/appex's (in which case we only disable code sharing, the build will still succeed), but when the same assembly is referenced multiple times from different paths in the same app/appex (which should be impossible, and if you look at the corresponding tests you'll see that I couldn't come up with a test case that triggered this error - but that doesn't mean customers won't find a way).

This is really just showing a better error if our assumptions (that it's impossible to reference multiple assemblies from different locations in the same app) don't hold true.